domain-models-runtime/src/main/java/org/folio/rest/persist/facets/FacetField.java

```diff
@@ -23,6 +23,7 @@ public FacetField(String path2facet, int amountOfValues2return){
     this(path2facet, amountOfValues2return, null);
   }
 
+  @SuppressWarnings("java:S5850")//Line:34 (in regex '$' has higher precedence than '|' , but it is a small regex so we can keep it)
```

Comment:
I would not suppress the warning but drop the first regexp and simply use the QUOTES_PATTERN regexp for both:

```
private static final Pattern QUOTES_PATTERN = Pattern.compile("'([^\']*)'");

  private FacetField(String path2facet, int topFacets2return, String alias){
    this.fieldPath = path2facet;
    this.topFacets2return = topFacets2return;
    if (path2facet == null || !path2facet.contains("\'")) {
      throw new ExceptionInInitializerError("Path to facet must be in the form of jsonb_field_name->'field1'->>'field2', for example jsonb->>'field'");
    }
    // get last occurrence of what is in between ''
    Matcher m = QUOTES_PATTERN.matcher(path2facet);
    while (m.find()) {
      this.alias = m.group(1);
    }
  }
```